### DISCUSSION TITLE
Voluntary Telemetry Submission

### DISCUSSION THREAD

**Discussion by JCWasmx86** [JCWasmx86/Swift-MesonLSP#104] [General]
## Why?
Sending telemetry has too many trade-offs:

- Infrastructure: I would need a server that receives the sent telemetry data. This costs money and requires engineering effort.
- Legal: There are privacy laws in the EU, but following them would require legal efforts of unknown dimension. As soon as you comment here, it's something else from a legal POV.
- UX: Nobody wants that data is sent to some random server. (Especially if it's a tool like a language server)

## What?
Please provide this telemetry data: (If it doesn't follow this format, I will edit it to make it it machine readable)
```
- OS: (E.g. Windows 10 | Windows 11 | macOS Sonoma | Fedora 40 | Archlinux | Ubuntu 22.04... etc.)
- Architecture: (One of: AArch64 | ARM32 | X86 | X86_64 | Other)
- Editor: (E.g. VSCode | Zed | GNOME Builder | neovim)
- Projects: (Optional) Project(s) you work on, with links to it (This allows to add them to regression tests)
- General Feedback: What do you like? What do you dislike? Do would you like to see?
```

**Thank you for your submissions**

**Comment by tristan957** [comment]
- OS: Fedora 40
- Architecture: x86_64
- Editor: Neovim
- Projects: github.com/postgres/postgres
- General Feedback: Might be cool to have a code action when you hover the argument of a dependency to add the wrap file under subprojects.

**Comment by 1ace** [comment]
- OS: Fedora 40
- Architecture: AArch64
- Editor: `vim` (+ `coc.nvim`)
- Projects: https://gitlab.freedesktop.org/mesa/mesa mostly
- General Feedback: *[haven't used it enough yet, so far so good but I'll come back and edit later once I have an opinion]*
